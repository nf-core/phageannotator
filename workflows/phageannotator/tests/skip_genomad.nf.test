nextflow_workflow {

    name "Test workflow: PHAGEANNOTATOR"
    script "workflows/phageannotator/main.nf"
    workflow "PHAGEANNOTATOR"

    // Dependencies
    tag "SEQKIT_SEQ"
    tag "QUALITYFILTERVIRUSES"
    tag "ANICLUSTER_ANICALC"
    tag "ANICLUSTER_ANICLUST"
    tag "ANICLUSTER_EXTRACTREPS"
    tag "COVERM_CONTIG"
    tag "FASTA_VIRUS_QUALITY_CHECKV"
    tag "FASTA_ALL_V_ALL_BLAST"
    tag "CAT_CAT"
    tag "BOWTIE2_BUILD"
    tag "GUNZIP"
    tag "FASTQ_ALIGN_BOWTIE2"


    test("Parameters: skip_genomad = true") {
        when {
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test' ],
                        [
                            file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_1.fastq.gz', checkIfExists:true),
                            file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_2.fastq.gz', checkIfExists: true)
                        ]
                    ],
                    [
                        [ id:'test2' ],
                        [
                            file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_1.fastq.gz', checkIfExists:true),
                            file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_2.fastq.gz', checkIfExists: true)
                        ]
                    ]
                )
                input[1] = Channel.of(
                    [
                        [ id:'test' ],
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz', checkIfExists:true)
                    ],
                    [
                        [ id:'test2' ],
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz', checkIfExists:true)
                    ]
                )
                """
            }
            params {
                outdir                      = "$outputDir"
                skip_genomad                = true
                // reduce hits, since reduction won't take place during genomad
                mash_screen_min_score       = 1
                mash_screen_winner_take_all = true
                // speed up options since tools are fully tested in subworkflows
                checkv_minimal_db           = true
                checkv_min_length           = 25000
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() },
                { assert path("${outputDir}/VirusAbundance/coverm/contig/all_samples_alignment_results.tsv").readLines().size() == 9 }
            )
        }
    }
}
