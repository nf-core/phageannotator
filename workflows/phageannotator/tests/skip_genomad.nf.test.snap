{
    "Parameters: skip_genomad = true": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    [
                        {
                            "id": "test2"
                        },
                        "quality_summary.tsv:md5,2a3e06d2323ddac0ed36e569d0ba1245"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "quality_summary.tsv:md5,2a3e06d2323ddac0ed36e569d0ba1245"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2.filtered.fna.gz:md5,3f978842932ad596be2b9987907077cb"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.fna.gz:md5,3f978842932ad596be2b9987907077cb"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_anicluster_representatives.fna.gz:md5,3f978842932ad596be2b9987907077cb"
                    ]
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "7": [
                    
                ],
                "8": [
                    
                ],
                "9": [
                    "versions.yml:md5,005aa3fca95c916b586f09b1c12aae6f",
                    "versions.yml:md5,005aa3fca95c916b586f09b1c12aae6f",
                    "versions.yml:md5,04bf25e5231ce1f8077581bb7a2c07f4",
                    "versions.yml:md5,0c310e8a973612ff99e0df847fb656ea",
                    "versions.yml:md5,0e57d76cb0b6149bbb32aded3595de9b",
                    "versions.yml:md5,0e57d76cb0b6149bbb32aded3595de9b",
                    "versions.yml:md5,1439e948f383899aaccbf49506923b04",
                    "versions.yml:md5,374c63bf7c31c6c34acbf8d74e6c7dcc",
                    "versions.yml:md5,74b2068d24b17a2f4a097a3063da3dae",
                    "versions.yml:md5,769114e3bb009aceb4763a1a0fa80d5a",
                    "versions.yml:md5,7cb8c67c6ff9cf01f799d6730d5ebd15",
                    "versions.yml:md5,7e54b5f822df67b99faf67629e81963f",
                    "versions.yml:md5,7e54b5f822df67b99faf67629e81963f",
                    "versions.yml:md5,8aebba726df29aad131772449019ee78",
                    "versions.yml:md5,987d2e54a85311b917599dc41c90418c",
                    "versions.yml:md5,a856eefde99260259440d942c1d56ff4",
                    "versions.yml:md5,b6a0d868f5f2be1093be05fc0638008d",
                    "versions.yml:md5,b6a0d868f5f2be1093be05fc0638008d",
                    "versions.yml:md5,bcf62c8c784b9f46c8372e86654cae6b",
                    "versions.yml:md5,bcf62c8c784b9f46c8372e86654cae6b",
                    "versions.yml:md5,d1e2a23440cc3dbeedc1c628bf792072",
                    "versions.yml:md5,d1e2a23440cc3dbeedc1c628bf792072"
                ],
                "anicluster_reps_fna_gz": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_anicluster_representatives.fna.gz:md5,3f978842932ad596be2b9987907077cb"
                    ]
                ],
                "filtered_viruses_fna_gz": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2.filtered.fna.gz:md5,3f978842932ad596be2b9987907077cb"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.fna.gz:md5,3f978842932ad596be2b9987907077cb"
                    ]
                ],
                "host_predictions_tsv": [
                    
                ],
                "instrain_gene_info": [
                    
                ],
                "marker_taxonomy_tsv": [
                    
                ],
                "pharokka_output_tsv": [
                    
                ],
                "versions": [
                    "versions.yml:md5,005aa3fca95c916b586f09b1c12aae6f",
                    "versions.yml:md5,005aa3fca95c916b586f09b1c12aae6f",
                    "versions.yml:md5,04bf25e5231ce1f8077581bb7a2c07f4",
                    "versions.yml:md5,0c310e8a973612ff99e0df847fb656ea",
                    "versions.yml:md5,0e57d76cb0b6149bbb32aded3595de9b",
                    "versions.yml:md5,0e57d76cb0b6149bbb32aded3595de9b",
                    "versions.yml:md5,1439e948f383899aaccbf49506923b04",
                    "versions.yml:md5,374c63bf7c31c6c34acbf8d74e6c7dcc",
                    "versions.yml:md5,74b2068d24b17a2f4a097a3063da3dae",
                    "versions.yml:md5,769114e3bb009aceb4763a1a0fa80d5a",
                    "versions.yml:md5,7cb8c67c6ff9cf01f799d6730d5ebd15",
                    "versions.yml:md5,7e54b5f822df67b99faf67629e81963f",
                    "versions.yml:md5,7e54b5f822df67b99faf67629e81963f",
                    "versions.yml:md5,8aebba726df29aad131772449019ee78",
                    "versions.yml:md5,987d2e54a85311b917599dc41c90418c",
                    "versions.yml:md5,a856eefde99260259440d942c1d56ff4",
                    "versions.yml:md5,b6a0d868f5f2be1093be05fc0638008d",
                    "versions.yml:md5,b6a0d868f5f2be1093be05fc0638008d",
                    "versions.yml:md5,bcf62c8c784b9f46c8372e86654cae6b",
                    "versions.yml:md5,bcf62c8c784b9f46c8372e86654cae6b",
                    "versions.yml:md5,d1e2a23440cc3dbeedc1c628bf792072",
                    "versions.yml:md5,d1e2a23440cc3dbeedc1c628bf792072"
                ],
                "virus_classification_tsv": [
                    
                ],
                "virus_enrichment_tsv": [
                    
                ],
                "virus_quality_tsv": [
                    [
                        {
                            "id": "test2"
                        },
                        "quality_summary.tsv:md5,2a3e06d2323ddac0ed36e569d0ba1245"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "quality_summary.tsv:md5,2a3e06d2323ddac0ed36e569d0ba1245"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-17T13:34:20.139289774"
    }
}